### ISSUE TITLE
EncodeException "The serialized GlobalReference has type PointerType ...

### ISSUE THREAD

**Issue by nwtgck** [llvm-hs/llvm-hs-quote#15]
Hi, developers of llvm-hs-quote.

## Environment

`llvm-hs-quote`:  e2dec47ad2bbf07d6a7d0e0cd8f91ae6c5186d72
stack solver: lts-10.3 (this is the same as `llvm-hs-quote`)

## Error

I had a error with the following code.

```hs
module10 :: AST.Module
module10 = [Quote.LLVM.llmod|
  define void @f() {
  entry:
    call void @g()
    ret void
  }
  define void @g() {
  entry:
    ret void
  }
|]

toLLVM :: AST.Module -> IO ()
toLLVM mod = Context.withContext $ \ctx -> do
  llvm <- Module.withModuleFromAST ctx mod Module.moduleLLVMAssembly
  BS.putStrLn llvm

main = toLLVM module10
```

I think the error occurs in `call void @g()`.

## Error message

```txt
 EncodeException "The serialized GlobalReference has type PointerType {pointerReferent = FunctionType {resultType = VoidType, argumentTypes = [], isVarArg = False}, pointerAddrSpace = AddrSpace 0} but should have type FunctionType {resultType = VoidType, argumentTypes = [], isVarArg = False}"
```

**Comment by cocreature** [comment]
Should be fixed in 1497273bf70ae22b76d190bcbc24737cfb50e334

**Comment by nwtgck** [comment]
Thank you very much for your quick fixing! You solved my problem.
